### PR TITLE
Make server-admins creation migration idempotent.

### PR DESCRIPTION
**Pull request by tylercloke**
If server-admins has an entry in erchef, skip its creation and continue to permission adds.

Fixes https://github.com/chef/chef-server/issues/822

[Build](http://wilson.ci.chef.co/job/chef-server-12-trigger-ad_hoc/208/downstreambuildview/)

### CODE REVIEW

File: src/chef-mover/src/mover_server_admins_global_group_callback.erl [hunk 1]
```diff
@@ -47,20 +48,33 @@ migration_init() ->
...
-        {chef_sql, {GroupError, _}} ->
-	    lager:error("Could not create new erchef group for server-admins."),
-            throw(GroupError);
-	_ -> true
-    end,
+
+    %% if server-admins is already created, skip this step
+    ServerAdminsAuthzId =
+        case get_server_admins_authz_group_if_exists() of
+            ExistingServerAdminsAuthzId ->
```
Comment (tylercloke) [inline src/chef-mover/src/mover_server_admins_global_group_callback.erl]: Er...does this need to go below the `needs_created` atom to match? Seems like it's matching `needs_created` based on my testing.

Comment (stevendanna) [inline src/chef-mover/src/mover_server_admins_global_group_callback.erl]: Yes, since this isn't bound at this point, it'll match everything.

Comment (tylercloke) [inline src/chef-mover/src/mover_server_admins_global_group_callback.erl]: That's what I thought. Fixed. Surprised I didn't get a compile warning, or maybe I just missed it.

**Comment by rmoshier** [comment]
LGTM 👍

**Comment by tylercloke** [comment]
Tested it out. Should be g2g.

**Comment by srenatus** [comment]
LGTM

**Pull request merged**